tests/unit_tests/audit_log/test_audit_query_handler.py

```diff
@@ -0,0 +1,127 @@
+from dataclasses import dataclass
+from typing import Generic
+from unittest.mock import Mock
+
+from exasol.analytics.audit.audit_query_handler import (
+    AuditQueryHandler,
+    ParameterType,
+    ResultType,
+)
+from exasol.analytics.query_handler.query.select import (
+    DbOperationType,
+    ModifyQuery,
+    Query,
+    SelectQueryWithColumnDefinition,
+)
+from exasol.analytics.query_handler.query_handler import QueryHandler
+from exasol.analytics.query_handler.result import (
+    Continue,
+    Finish,
+)
+from exasol.analytics.schema import (
+    Column,
+    decimal_column,
+)
+from exasol.analytics.schema.schema_name import SchemaName
+from exasol.analytics.schema.table_name import TableName
+from exasol.analytics.schema.table_name_impl import TableNameImpl
+from tests.utils.audit_table_utils import (
+    QueryStringCriterion,
+    assert_queries_match,
+    create_insert_query,
+    expected_query,
+)
+
+
+@dataclass
+class MyParameterType(Generic[ParameterType]):
+    db_schema: str
+    table_name_prefix: str
+
+
+AUDIT_TABLE_NAME_PREFIX = "PPP"
+
+AUDIT_TABLE_NAME = TableNameImpl(
+    f"{AUDIT_TABLE_NAME_PREFIX}_AUDIT_LOG", SchemaName("SSS")
+)
+
+
+def create_audit_query_handler(
+    child: QueryHandler[ParameterType, ResultType],
+) -> AuditQueryHandler:
+    def schema_getter(parameter: MyParameterType) -> str:
+        return parameter.db_schema
+
+    def table_name_prefix_getter(parameter: MyParameterType) -> str:
+        return parameter.table_name_prefix
+
+    def additional_columns_provider(parameter: MyParameterType) -> list[Column]:
+        return [decimal_column("DDD", precision=9)]
+
+    parameter = MyParameterType(
+        db_schema="SSS",
+        table_name_prefix=AUDIT_TABLE_NAME_PREFIX,
+    )
+    return AuditQueryHandler(
+        parameter=parameter,
+        context=Mock(),
+        query_handler_factory=Mock(return_value=child),
```

Comment:
not sure, if using a Mock here is useful, because you don't use the Mock for testing. A lambda or internal function would do the job too

Reply:
`Mock()` requires the least amount of code, here. 
Also, the constructor will call `context.get_child_query_handler_context()`. 
So, an internal function or lambda is not sufficient from my perspective. 

Please correct me if I'm wrong.

Reply:
See next push.